pkg/descheduler/strategies/duplicates.go

```diff
@@ -42,77 +42,71 @@ func RemoveDuplicatePods(
 	client clientset.Interface,
 	strategy api.DeschedulerStrategy,
 	nodes []*v1.Node,
-	evictLocalStoragePods bool,
 	podEvictor *evictions.PodEvictor,
 ) {
 	for _, node := range nodes {
 		klog.V(1).Infof("Processing node: %#v", node.Name)
-		duplicatePods := listDuplicatePodsOnANode(ctx, client, node, strategy, evictLocalStoragePods)
-		for _, pod := range duplicatePods {
-			if _, err := podEvictor.EvictPod(ctx, pod, node); err != nil {
-				klog.Errorf("Error evicting pod: (%#v)", err)
-				break
-			}
+		pods, err := podEvictor.ListEvictablePodsOnNode(ctx, client, node)
```

Comment:
this diff for this strategy looks a lot more complicated than it is. I just moved the private `listDuplicatePodsOnANode` logic (~30 lines) into the main public function. If anyone has any objections I'll revert it but didn't really see that it was necessary to break it up like that